### DISCUSSION TITLE
네이버 상품 상세정보의 모바일 미리보기 생성의 기준

### DISCUSSION THREAD

**Discussion by importunate-dev** [commerce-api-naver/commerce-solutions#96] [묻고 답하기]
안녕하세요.
네이버 상품 상세정보 내에 `<a>` 태그가 포함된 이미지 태그를 추가하였습니다.
이후 모바일에서 보는 화면과 PC로 보는 상품 상세보기 페이지가 달라지는 이슈가 발생했습니다.
저희가 추측하기로는 `<a>` 태그가 원인인 것으로 추측하고 있는데, 이미지 태그의 src 내에 외부 링크(예. S3 링크) 등이 포함되는 것도
원인이 될 수 있을 까요?

1. 모바일에서도 네이버 상품 상세정보가 원본과 동일하게 보이기 위하여 사용해선 안되는 태그나 조건이 구체적으로 있는지 궁금합니다.
2. 특히, 저희가 이 `<a>` 태그가 포함된 `<img>` 태그를 추가한 이유 모바일에서 상품 상세정보 페이지에 '원본 보기' 버튼이 추가되었는데 이것이 생기는 기준 또한 궁금합니다.
3. 혹시 모바일 미리보기 페이지가 있는 경우, 상품 상세정보 페이지가 변동된 후에 모바일 미리보기 페이지가 업데이트 되는 데까지 걸리는 시간도 궁금합니다.
4. 현재 상품정보에서 detailContent 를 조회하여 상품 상세정보를 보고있는데, 상세정보가 스마트에디터 원으로 작성한건지 직접 HTML로 작성한건지 알 수 있는 방법이 있는지 궁금합니다.
5. 스마트에디터원으로 작성한 상세페이지를 수정할 경우, HTML로 변경되는데 이를 유지할 방법은 현재 없는지 궁금합니다.

**Comment by commerce-api-naver** [comment]
안녕하세요. 커머스API 이민석입니다.

### 1, 2번째 문의에 답변
'모바일미리보기'는 커머스API로 제어하는 영역이 아니며 스마트스토어 서비스 내부에서 돌아가는 내용이기에 여기서 정확한 확인을 도와드리긴 어렵습니다.
단, 모바일미리보기는 판매자 분께서 입력해주신 상품상세정보 내용에 대한 스냅샷 형태로 관리됩니다. 스냅샷 생성 과정에서 Bot 접근 차단 등으로 일부 이미지나 상품상세정보 HTML을 구성하는 요소에 접근이 불가능할 경우 스냅샷 생성에 실패할 수 있습니다.
이러한 경우 원본 표시로 처리될 수 있으며 상세한 실패 사유는 스마트스토어센터 내 '모바일미리보기 생성 실패 사유'를 참고해주시는 것을 권해드리고 있습니다.
- 관련한 스마트스토어 고객센터 도움말: https://help.sell.smartstore.naver.com/faq/content.help?faqId=3582

### 3번째 문의에 답변
통상적으로 상품상세정보 갱신 후 최소 수 시간에서 최대 24시간까지 소요될 수 있습니다.

### 4번째 문의에 답변
상품상세정보의 '스마트에디터 ONE' 작성 시 `detailContent` 필드 값으로는 '스마트에디터 ONE'으로 작성해주신 상세설명의 HTML 값만 제공되며 '스마트에디터 ONE' 고유 포맷과 관련한 각종 메타 데이터는 커머스API로는 제공하지 않습니다.
그렇기에 현재 커머스API에서 제공중인 데이터만으로는 정확하게 상품상세정보의 포맷 유형(스마트에디터 ONE/HTML 작성)을 구분하긴 어렵습니다.

#### HTML 주석 Prefix를 이용한 추정 방법
'스마트에디터 ONE' 작성 시 `detailContent` 필드 값으로 제공하는 HTML 내용에는 특유의 공통된 Prefix를 가진 HTML 주석을 확인하실 수 있습니다.
이러한 특유의 패턴을 근거로 대상 내용의 '스마트에디터 ONE' 작성 여부를 유추할 순 있으나 똑같은 HTML 주석 내용을 'HTML 직접 입력'으로도 입력할 수 있기에 이러한 방법은 어느정도의 추정 용도로만 사용하는 것을 권해드리고 있습니다.

### 5번째 문의에 답변
상품 수정 시 `detailContent` 필드 값은 생략이 가능합니다.
이때 `detailContent` 필드 자체를 JSON 노드에 포함하지 않는 방법(_null_)으로 생략할 경우 저장된 상품상세정보 내용이 그대로 보존됩니다.
이 기능을 이용할 경우 상품 정보를 수정하더라도 '스마트에디터 ONE' 포맷을 보존하여 다른 상품 정보를 수정할 수 있습니다.
- 관련한 자주 묻는 질문: https://github.com/commerce-api-naver/commerce-api/discussions/26

> [!NOTE]
> 위 링크는 '내 스토어 애플리케이션 기술지원 공간(`/commerce-api`)'로 연결됩니다.
> 커머스솔루션 연동 개발 문의는 반드시 '커머스솔루션 개발사 기술지원 공간(`/commerce-solutions`)'으로 남겨주실 수 있도록 문의 작성 시 좌측 상단의 레파지토리 명을 확인해보시기 바랍니다.
